crates/codegen/syntax_templates/src/shared/cst.rs

```diff
@@ -52,4 +53,34 @@ impl Node {
             Self::Token(node) => (&node.text).into(),
         }
     }
+
+    pub fn cursor(&self) -> Cursor {
+        Cursor::new(self.clone())
+    }
+
+    pub fn is_token_kind(&self, kinds: &[TokenKind]) -> Option<&Rc<TokenNode>> {
+        if let Node::Token(token_node) = self {
+            if kinds.contains(&token_node.kind) {
+                return Some(token_node);
+            }
+        }
+        return None;
+    }
+
+    pub fn is_token_matching<F: Fn(&Rc<TokenNode>) -> bool>(&self, predicate: F) -> bool {
```

Comment:
Is `is_token_matching` and `is_rule_matching` intended to be shared between languages? what is the use case here?

Reply:
Yes.

The use case if to supply a predicate, which is passed the TokenNode or RuleNode i.e. pre-matched. For use when traversing, and for alignment with the Cursor API